app/instrumentation-tests/src/org/commcare/utils/InstrumentationUtility.kt

```diff
@@ -310,6 +318,18 @@ object InstrumentationUtility {
         }
     }
 
+    fun verifyIntentTextOnSubmitReport(text : String){
+        var intentcallback = IntentCallback { intent ->
+            var extraText = intent.extras!!.getString(Intent.EXTRA_TEXT)
+            assertTrue(extraText!!.contains(text))
+        }
```

Comment:
you can also add a check for intent action like `assertTrue(intent.action.contentEquals(Intent.ACTION_SEND))`

Reply:
I tried adding the assertion for checking the action but it kept failing as it returned a different intent (CHOOSER). So didn't add the same.